config_role.go

```diff
@@ -0,0 +1,96 @@
+package evergreen
+
+import (
+	"time"
+
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// LDAPRoleMap contains a mapping of LDAP groups for a user to roles. LDAP
+// groups are represented by their name and roles by their unique ID in the
+// roles collection.
+type LDAPRoleMap struct {
+	LastUpdated time.Time         `bson:"last_updated" json:"last_updated" yaml:"last_updated"`
+	Map         map[string]string `bson:"map" json:"map" yaml:"map"`
+}
+
+var (
+	ldapRoleMapLastUpdatedKey = bsonutil.MustHaveTag(LDAPRoleMap{}, "LastUpdated")
+	ldapRoleMapMapKey         = bsonutil.MustHaveTag(LDAPRoleMap{}, "Map")
+)
+
+// SectionId returns the ID for this section of the admin settings.
+func (m *LDAPRoleMap) SectionId() string { return "ldap_role_map" }
+
+// Get queries the database for the LDAP-role map.
+func (m *LDAPRoleMap) Get(env Environment) error {
+	ctx, cancel := env.Context()
+	defer cancel()
+	coll := env.DB().Collection(ConfigCollection)
+
+	res := coll.FindOne(ctx, byId(m.SectionId()))
+	if err := res.Err(); err != nil {
+		return errors.Wrapf(err, "error retrieving section %s", m.SectionId())
+	}
+
+	if err := res.Decode(m); err != nil {
+		if err == mongo.ErrNoDocuments {
+			*m = LDAPRoleMap{}
+			return nil
+		}
+		return errors.Wrapf(err, "error retrieving section %s", m.SectionId())
+	}
+
+	return nil
+}
+
+// Add adds a new (or updates an existing) LDAP group to role mapping in the
+// database.
+func (m *LDAPRoleMap) Add(group, roleID string) error {
```

Comment:
This doesn't implement the ConfigSection interface so I don't think the existing code will update this (I also think you need to set the "id" tag)